tutorials/learn_digital_audio/01_Intro.cpp

```diff
@@ -0,0 +1,124 @@
+// Joel A. Jaffe 2024-07-17
+// With acknowledgement and gratitude to 
+// Karl Yerkes, Graham Wakefield, and Mahdi Ayman
+
+/* Intro
+This series is intended as an introduction to 
+digital audio programming using AlloLib.
+
+This first app will demonstrate how to create a basic 'AlloApp'
+and also introduce the oscilloscope, an invaluable tool for 
+visualizing audio data.
+*/
+
+// #includes allow recursive inclusion of classes, functions, etc
+// they allow us to "build on the shoulders of giants" and develop things incrementally 
+#include "al/app/al_App.hpp"
+using namespace al; // use the namespace, so that we don't have to prepend objects with `al::`
+
+/**
+ * @brief An oscilloscope class that inherits from `al::Mesh`.
+ * Useful for visualizing audio data. 
+ * Templated so that different types of audio data (float, double, etc) can be used.
+ */
+template <typename T>
+class Oscilloscope : public al::Mesh { // inherit from `al::Mesh`, the al:: is optional when using namespace
+private: // private 'member' variables unique to each instance 
+  int bufferSize; // how many samples in the past oscilloscope will remember
+  int writeIndex; // variable that tracks what index of the buffer to write to
```

Comment:
writeIndex is not initialized and may trigger a segfault in writeSample. Suggest int writeIndex=0;

Reply:
Thanks for the feedback Lance! I'm a huge fan of your work. I will try to address each issue as commented individually.